MediaBrowser.Providers/Manager/ItemImageProvider.cs

```diff
@@ -61,6 +61,30 @@ public ItemImageProvider(ILogger logger, IProviderManager providerManager, IFile
             _fileSystem = fileSystem;
         }
 
+        /// <summary>
+        /// Removes all existing images from the provided item.
+        /// </summary>
+        /// <param name="item">The <see cref="BaseItem"/> to remove images from.</param>
+        /// <returns><c>true</c> if changes were made to the item; otherwise <c>false</c>.</returns>
+        public bool RemoveImages(BaseItem item)
+        {
+            var singular = new List<ItemImageInfo>();
+            for (var i = 0; i < _singularImages.Length; i++)
+            {
+                var currentImage = item.GetImageInfo(_singularImages[i], 0);
+                if (currentImage != null)
+                {
+                    singular.Add(currentImage);
+                }
+            }
+
+            var oldBackdropImages = item.GetImages(ImageType.Backdrop).ToArray();
+            var toRemove = singular.Concat(oldBackdropImages).ToArray();
+            PruneImages(item, toRemove);
```

Comment:
```suggestion
            singular.AddRange(item.GetImages(ImageType.Backdrop);
            PruneImages(item, singular);
```

and change PruneImages to be
```patch
Index: MediaBrowser.Providers/Manager/ItemImageProvider.cs
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/MediaBrowser.Providers/Manager/ItemImageProvider.cs b/MediaBrowser.Providers/Manager/ItemImageProvider.cs
--- a/MediaBrowser.Providers/Manager/ItemImageProvider.cs	(revision 6d299eed495549fd92402571577d732c74a73b37)
+++ b/MediaBrowser.Providers/Manager/ItemImageProvider.cs	(date 1641255052822)
@@ -340,9 +340,9 @@
             return options.IsEnabled(type);
         }
 
-        private void PruneImages(BaseItem item, ItemImageInfo[] images)
+        private void PruneImages(BaseItem item, IReadOnlyList<ItemImageInfo> images)
         {
-            for (var i = 0; i < images.Length; i++)
+            for (var i = 0; i < images.Count; i++)
             {
                 var image = images[i];
```